hivemind/utils/mpfuture.py

```diff
@@ -46,6 +48,7 @@ class MPFuture(base.Future, Generic[ResultType]):
     Any process can access future status and set the result / exception and check for state.
     However, only the original process (i.e. the process that created the future) can await the result or exception.
 
+    :param synchronize: if True (default), future will request state from origin, otherwise it will only use local state
```

Comment:
nit: Please clarify here that it can be used if the future is guaranteed to never be cancelled from the origin side and leads to a minor speedup.